plugins/modules/packaging/os/yum_versionlock.py

```diff
@@ -125,23 +126,20 @@ def main():
     if state in ('present'):
         command = 'add'
         for single_pkg in packages:
-            if single_pkg not in versionlock_packages:
-                if module.check_mode:
-                    changed = True
-                    continue
+            if not any(fnmatch(pkg.split(":", 1)[-1], single_pkg) for pkg in versionlock_packages.split()):
                 packages_list.append(single_pkg)
```

Comment:
Shouldn't the matching packages be added to this list instead of the (potential) pattern? (Same below.)

Reply:
I'm not sure if I understood what you mean, `packages` is filled in from module input, and `single_pkg` is each entry of that input, that's what is passed into `yum versionlock add/delete $single_pkg`.

Reply:
Why not simply filter the list of packages and provide that to `yum versionlock`? Right now, the code is very much dependent on that `fnmatch` does exactly the same package name matching as `yum` itself does. Also when you precompute the list of packages to (un-)lock, the module can return for which ones changes were made (which also makes it easier to debug).